core/corehttp/p2p_proxy.go

```diff
@@ -60,12 +61,16 @@ func parseRequest(request *http.Request) (*proxyRequest, error) {
 		return nil, fmt.Errorf("Invalid request path '%s'", path)
 	}
 
+	if _, err := peer.Decode(split[2]); err != nil {
+		return nil, fmt.Errorf("Invalid request path '%s'", path)
+	}
```

Comment:
This effectively changes the error code for an invalid peerID from 502 to 400 and shows an error to the user instead of just logging on the server.